### PR TITLE
feat(exec): add --detach flag and exec logs subcommand

### LINKED ISSUE #187: feat: Add detach mode and result retrieval command
## Background

When running a long-running command (e.g., `apt upgrade`, `mysqldump`) via `alpacon exec`, the CLI disconnects after a 5-minute polling timeout while alpamon on the server side continues executing. To check the result, the user must log into the server directly every time.

The current `PollCommandExecution()` has a fixed 5-minute timeout with 1-second polling intervals, and there is no `--detach` / `--follow` flag.

## Goal

- Resolve the mismatch between the CLI timeout and the server-side command completion time
- Deployable immediately by modifying only the CLI without any server changes

---

### 1. Add `--detach` flag

`alpacon exec <command> --detach`

- Submit the command, print the job ID, and exit immediately
- Returns without polling
- Example output:
  ```
  Job submitted: a1b2c3d4-...
  Run `alpacon exec logs a1b2c3d4-...` to check the result.
  ```

### 2. Add `alpacon exec logs <job-id>` command

- Fetch results via `GET /api/events/commands/{id}/`
- If completed: print the result and exit
- If still running: print the current status and exit (or wait with polling using `--wait` flag)

### 3. Adjust polling timeout

- Increase the default 5-minute timeout to at least 30 minutes, or make it configurable
- Allow override via environment variable or config file

---

## Files to change (expected)

| File | Change |
|------|--------|
| `cmd/exec/exec.go` | Add `--detach` flag |
| `cmd/exec/run.go` | Handle detach branch, adjust polling timeout |
| `cmd/exec/logs.go` | New — `logs <job-id>` subcommand |
| `api/event/event.go` | Parameterize `PollCommandExecution()` timeout |

---

## Acceptance criteria

- Running `alpacon exec "apt upgrade" --detach` prints the job ID and exits immediately
- Running `alpacon exec logs <job-id>` prints the result of a completed command
- Running `logs` against an in-progress command displays the current status

### PR DESCRIPTION
**Pull request by jisung-02**
## Overview

Adds `--detach` flag and `exec logs` subcommand to `alpacon exec`.
Commands can now be submitted asynchronously and their results retrieved later by job ID.

## Usage

**Submit a command (returns immediately)**
```
alpacon exec --detach web-editor -- apt-get update
# stdout: Job submitted: <UUID>
# stderr: Run `alpacon exec logs <UUID>` to check the result.
```

**Retrieve the result**
```
alpacon exec logs <UUID>
```

- **Completed**: command output printed to stdout; exits with the remote exit code
- **Still running**: prints "command is still running (status: ...)" to stderr; exits 0
- **Failed** (stuck/error/cancelled): prints error message to stderr; exits 1

**JSON output**
```
alpacon exec --output json --detach web-editor echo hello
# {"job_id": "<UUID>"}
```

> **Note**: `--detach` and other flags must appear before the server name, consistent with `exec [flags] SERVER COMMAND` usage.

### New
- `--detach` flag: submit a command and return the job ID immediately
- `exec logs <JOB_ID>`: single-fetch subcommand to check status or retrieve result
- `event.GetCommandByID()`: fetch a single command by ID
- `event.SubmitCommand()`: extracted from `RunCommand()` for reuse

### Improvements
- `IsRunningStatus()` exported: single source of truth for in-progress status classification, shared by polling and `exec logs`
- Default exec timeout raised to 30 minutes; `ALPACON_EXEC_TIMEOUT` env var allows override
- `cancelled` status now handled (was previously missing)
- Unknown terminal statuses treated as exit 1 consistently in both `RunCommand` and `logsCommandOutcome`
- Fixed timer-reset race: drain `timer.C` before `Reset` to prevent spurious `ClientTimeoutError`
- WorkSession exit code preserved in the detach path

## Tests

- `TestIsRunningStatus`: verifies status classification for all known values
- `TestLogsCommandOutcome`: 13 cases covering completed, running, stuck, cancelled, error, exit code propagation, unknown status
- `TestRunCommand_UnrecognisedTerminalStatusReturnsError`: ensures unknown terminal statuses return an error
- `TestParseRemoteExecArgs`: validates `--detach` and `--detach=VALUE` flag parsing

## Smoke test (manual, against `web-editor`)

```
# 1. Submit detached command
$ ./alpacon exec --detach web-editor echo "hello detach"
Using work-session affcfa45-fd1b-4135-908d-7947c9d3c4d8
Job submitted: d1000d7c-fd44-4206-9143-579e6dab63df
Run `alpacon exec logs d1000d7c-fd44-4206-9143-579e6dab63df` to check the result.

# 2. Retrieve result (completed)
$ ./alpacon exec logs d1000d7c-fd44-4206-9143-579e6dab63df
hello detach

# 3. Submit a long-running command and check while running
$ ./alpacon exec --detach web-editor -- sleep 30
Job submitted: cf50c60e-2b50-4ccc-b66b-796ff51b3c08
$ ./alpacon exec logs cf50c60e-2b50-4ccc-b66b-796ff51b3c08
command is still running (status: running).
Run `alpacon exec logs cf50c60e-2b50-4ccc-b66b-796ff51b3c08` again to check later.

# 4. Failing command (false) — empty output, exit code 1
$ ./alpacon exec --detach web-editor -- false
Job submitted: 3cfce49f-ca9d-4255-ae2c-811d993503c2
$ ./alpacon exec logs 3cfce49f-ca9d-4255-ae2c-811d993503c2; echo $?
1

# 5. JSON output
$ ./alpacon exec --output json --detach web-editor echo "hello json" 2>/dev/null
{
  "job_id": "d0bbd7a8-40e2-47d6-82ca-df97a58bde63"
}
```

Closes #187

### CODE REVIEW

File: cmd/exec/exec.go [hunk 1]
```diff
@@ -92,6 +100,25 @@ Requires an active WorkSession when using Browser login (Auth0); Token auth (API
...
 
 		env := make(map[string]string)
+
+		if parsed.Detach {
+			resp, err := event.SubmitCommand(alpaconClient, parsed.Server, parsed.Command, parsed.Username, parsed.Groupname, env, workSessionID)
+			if err != nil {
+				utils.HandleWorkSessionError(err, "command", parsed.Server, authMethod, workSessionID)
+				utils.CliErrorWithExit("failed to submit command on '%s': %s", parsed.Server, err)
+				return
+			}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/exec.go:135]: Fixed. Inlined `HandleCommonErrors` directly in the `--detach` branch of `exec.go`, mirroring the MFA/username-required/token-refresh retry logic from the sync path. A separate `SubmitCommandWithRetry` helper was considered but skipped — the detach submit is called from one site only, so extraction would add indirection without reuse benefit.

File: cmd/exec/logs.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/alpacax/alpacon-cli/api/event"
+	"github.com/alpacax/alpacon-cli/client"
+	"github.com/alpacax/alpacon-cli/utils"
+	"github.com/spf13/cobra"
+)
+
+var LogsCmd = &cobra.Command{
+	Use:   "logs JOB_ID",
+	Short: "Fetch the result of a detached command",
+	Long: `Fetch the result of a command submitted with --detach.
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/logs.go]: Fixed. Renamed `LogsCmd` → `logsCmd` to match the unexported convention used throughout the repo (e.g. `restartAgentCmd`, `webshListCmd`). Registration in `init()` updated accordingly.

File: cmd/exec/logs.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+If the command is still running, prints the current status and exits.
+Run the command again later to check for completion.`,
+	Example: `  alpacon exec logs a1b2c3d4-5678-abcd-ef01-234567890abc`,
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		jobID := args[0]
+
+		alpaconClient, err := client.NewAlpaconAPIClient()
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/logs.go:33]: Fixed. Added `utils.IsUUID(jobID)` validation before the API call. Returns an immediate, clear error for non-UUID input and prevents `path.Join` canonicalization from turning values like `../secrets` into unintended request paths.

File: cmd/exec/parse_test.go [hunk 4]
```diff
@@ -603,6 +603,54 @@ func TestParseRemoteExecArgs_OutputFlagEmptyValue(t *testing.T) {
...
+			wantCommand: "ls",
+		},
+		{
+			name:        "--detach after server is treated as command arg",
+			args:        []string{"server", "--detach"},
+			wantDetach:  false,
+			wantServer:  "server",
+			wantCommand: "--detach",
+		},
+	}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/parse_test.go:642]: Fixed. Added a `--detach=true` equals-form test case (`TestParseRemoteExecArgs_DetachFlag`) to cover the `--detach=VALUE` parsing path that `ParseRemoteExecArgs` already accepted.

File: cmd/exec/parse.go [hunk 5]
```diff
@@ -94,6 +96,8 @@ func ParseRemoteExecArgs(args []string) RemoteExecArgs {
 			if outputFormat == "" {
 				return RemoteExecArgs{Err: "--output requires a value (table|json)"}
 			}
+		case arg == "--detach" || strings.HasPrefix(arg, "--detach="):
+			detach = true
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/parse.go]: Fixed. Removed support for the equals-form altogether: `--detach=VALUE` now returns an error ("--detach does not accept a value; use --detach alone"). Only bare `--detach` is accepted. The test case was moved from `TestParseRemoteExecArgs_DetachFlag` to `TestParseRemoteExecArgs_Errors` to reflect the new rejection behavior.

File: api/event/event_test.go [hunk 6]
```diff
@@ -458,6 +469,79 @@ func TestPollCommandExecution_TerminalStatusReturnsBeforeTimeout(t *testing.T) {
...
+
+	ac := &client.AlpaconClient{HTTPClient: ts.Client(), BaseURL: ts.URL}
+	_, err := GetCommandByID(ac, "job-abc-123")
+	require.Error(t, err)
+}
+
+func TestExecTimeout_Default(t *testing.T) {
+	t.Setenv("ALPACON_EXEC_TIMEOUT", "")
+	assert.Equal(t, 30*time.Minute, execTimeout())
+}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply api/event/event_test.go:540]: Fixed. `TestPollCommandExecution` now calls `pollCommandExecution` directly with an explicit 30s timeout and 5ms tick, removing the dependency on the env var entirely. All `TestRunCommand_*` functions now guard with `t.Setenv("ALPACON_EXEC_TIMEOUT", "")` at their top so a short env value cannot race against the first poll tick and cause spurious failures.

File: cmd/exec/exec.go [hunk 7]
```diff
@@ -92,6 +102,44 @@ Requires an active WorkSession when using Browser login (Auth0); Token auth (API
...
+					},
+				})
+			}
+			if err != nil {
+				utils.HandleWorkSessionError(err, "command", parsed.Server, authMethod, workSessionID)
+				utils.CliErrorWithExit("failed to submit command on '%s': %s", parsed.Server, err)
+				return
+			}
+			if utils.OutputFormat == utils.OutputFormatJSON {
+				data, _ := json.Marshal(map[string]string{"job_id": resp.ID})
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/exec.go]: Fixed. Now using `data, err := json.Marshal(...)` with explicit error handling before calling `utils.PrintJson`.

File: cmd/exec/logs.go [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+		}
+
+		alpaconClient, err := client.NewAlpaconAPIClient()
+		if err != nil {
+			utils.CliErrorWithExit("Connection to Alpacon API failed: %s. Consider re-logging.", err)
+			return
+		}
+
+		details, err := event.GetCommandByID(alpaconClient, jobID)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/logs.go:40]: Fixed. Now calling `config.ResolveAuthMethod()` before the API call and passing the result to `utils.HandleWorkSessionError` on the error path, consistent with other commands.

File: cmd/exec/logs.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+			os.Exit(exitCode)
+		}
+	},
+}
+
+func init() {
+	ExecCmd.AddCommand(logsCmd)
+}
+
+// logsCommandOutcome maps GetCommandByID details to (stdout, stderr, exitCode). stderrLine always ends with \n.
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/logs.go]: Fixed. Updated the comment to "If non-empty, stderrLine ends with \n".

File: cmd/exec/logs.go [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+		alpaconClient, err := client.NewAlpaconAPIClient()
+		if err != nil {
+			utils.CliErrorWithExit("Connection to Alpacon API failed: %s. Consider re-logging.", err)
+			return
+		}
+
+		details, err := event.GetCommandByID(alpaconClient, jobID)
+		if err != nil {
+			utils.HandleWorkSessionError(err, "command", "", authMethod, "")
+			utils.CliErrorWithExit("failed to fetch command result: %s", err)
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/exec/logs.go:42]: No change needed here. `exec logs` is a read-only GET request (`GetCommandByID`) that retrieves historical command results — it does not go through the command execution pipeline where WorkSession gating is applied. A WorkSession error on this path is not an expected scenario, so the empty `activeWS` is intentional. The `serverName` is also correctly left empty since `exec logs` takes a job UUID, not a server name. The generic `<ID>` placeholder in the extend suggestion is sufficient for this edge case.

**Review by eunyoung14 — APPROVED** [review]
Code reviewed — well-structured implementation with good test coverage. Merge conflict resolved and --detach examples added to help text.
